### ISSUE TITLE
Hot Reload with Elmish

### ISSUE THREAD

**Issue by nabeelvalley** [fsbolero/Bolero#187]
Good day, is it possible to configure hot-reloads when using F#/Elmish instead of HTML? 

If not are there plans to include this at some point - or is it not really possible?

**Comment by Tarmil** [comment]
There was @tylerhartwig's FCSWatch-based hot reload tool, but it hasn't been updated in quite a while: https://github.com/tylerhartwig/Elmish.HotReload.Bolero. It would definitely be worth looking into updating it to make it work with the latest Bolero and Blazor.

**Comment by tylerhartwig** [comment]
Yes, the effort was never fully completed, but it is possible to configure some amount of hot reload for the view/update/init parts of Elmish

**Comment by farlee2121** [comment]
Do we know if the hot reload changes for Blazor with work with Bolero?
https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#initial-net-hot-reload-support

It [looks like](https://github.com/dotnet/sdk/blob/d202ae1de2a0e9178cebfb5f51cb1b9a46baacfd/src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs#L66) it only handles C# files, but my understanding of how the hot reload works is pretty basic.

Side note: 
For any who coming looking for any kind of runtime reload, `dotnet watch run` will recompile while running, but requires a page refresh for changes to show.

**Comment by Tarmil** [comment]
> Do we know if the [hot reload changes for Blazor](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#initial-net-hot-reload-support) will work with Bolero?

It seems to be based on Edit & Continue, which only supports C# and VB unfortunately.

**Comment by farlee2121** [comment]
It looks like they [plan to support hot reload for F# along with .NET 6](https://devblogs.microsoft.com/dotnet/introducing-net-hot-reload/#supported-unsupported-changes-and-languages).

The post isn't specific about if this applies to the VS debugger, dotnet watch, and I wasn't able to track down the github issues where it might have been discussed. Still, fingers crossed

**Comment by Tarmil** [comment]
Not along with .NET 6, but "in a future release", so I don't expect it soon; but the more we talk about it, the more resources they'll dedicate to it and the quicker it will come.

**Comment by farlee2121** [comment]
Ah. I took that to mean a future release of .NET 6 previews. Your probably right though.
